app/services/course/assessment/submission_service.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+class Course::Assessment::SubmissionService < SimpleDelegator
+  # Pass the defined instance variables to the controller, so that views can access them.
+  # Call this method at the end of the action if there are any instance variables defined in the
+  # action.
+  def assign_instance_variables
+    instance_variables_to_assign.each do |name|
+      value = instance_variable_get(name)
+      @_controller.instance_variable_set(name, value)
+    end
+  end
+
+  def update
+    if @submission.update_attributes(update_params)
+      redirect_to edit_course_assessment_submission_path(current_course, @assessment, @submission),
+                  success: t('course.assessment.submissions.update.success')
+    else
+      render 'edit'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.